toolkit/tools/pkg/licensecheck/testdata/generate_test_data.py

```diff
@@ -0,0 +1,139 @@
+    output = output.split("\n")
+    output = [url.strip() for url in output if not url.endswith(".src.rpm")]
+    output = [url for url in output if url]
+
+    return output
+
+# query_rpm_url() runs the 'rpm' command with the given query and URL. It returns a list of files based on the query.
+def query_rpm_url(out_file: str, args: list[str]) -> list[str]:
+    cmd = ["rpm"] + args + [out_file]
+    # Run the bash script and capture the output.
+    output = subprocess.check_output(cmd, text=True, stderr=subprocess.DEVNULL)
+
+    # If the output has the string '(contains no files)', then there are no files to list
+    if "(contains no files)" in output:
+        return []
+
+    output = output.split("\n")
+    output = [file for file in output if file]
+    return output
+
+def get_name(out_file: str) -> str:
+    return query_rpm_url(out_file, ["-q", "--qf", "%{NAME}\n"])
+
+def get_license_files(out_file: str) -> list[str]:
+    return query_rpm_url(out_file, ["-qL"])
+
+def get_doc_files(out_file: str) -> list[str]:
+    return query_rpm_url(out_file, ["-qd"])
+
+def get_all_files(out_file: str, filter_list: list[str]) -> list[str]:
+    all_files_and_dirs = query_rpm_url(out_file, ["-q", "--qf", "[%{FILEMODES:perms} %{FILENAMES}\n]"])
+    # Each line will be in the format "drwxr-xr-x /a/directory" or "-rw-r--r-- /a/directory/a_file", remove the
+    # directories and keep the files, then drop the permissions part of the string.
+    all_files = [file.split(' ', 1)[1] for file in all_files_and_dirs if file[0] != "d"]
+    filter_set = set(filter_list)
+    filtered_files = [file for file in all_files if file not in filter_set]
+    return filtered_files
+
+# get_files_for_url() returns a result object with the URL, license files, and doc files for a given URL.
+def get_files_for_url(url: str) -> dict:
+    # Get a tempdir to hold the rpm in so we can query it
+    with tempfile.TemporaryDirectory() as tempdir:
+        # Download the file to the tempdir
+        out_file = os.path.join(tempdir, "pkg.rpm")
+        urllib.request.urlretrieve(url, out_file)
+        license_files = get_license_files(out_file)
+        doc_files = get_doc_files(out_file)
+        all_other_files = get_all_files(out_file, license_files + doc_files)
+        res = {
+            "url": url,
+            "pkg_name": get_name(out_file)[0],
+            "license_files": license_files,
+            "doc_files": doc_files,
+            "all_other_files": all_other_files
+
+        }
+    return res
+
+# Write the results to a file.
+def write_to_file(file_list: list[(str,str)], output_file: str):
+    print(f"Writing to {output_file}")
+    file_list.sort()
+    with open(output_file, "w") as f:
+        for pkg_name, full_path_list  in file_list:
+            for full_path in full_path_list:
+                f.write(f"{pkg_name}`{full_path}\n")
+
+# Put the debug info packages first since they tend to be really big,
```

Comment:
It is generally a good idea to always use the Python's main func pattern:

```python
def main():
   ...

if __name__ == '__main__':
    main()
```

Reply:
Done